src/test/java/kitchenpos/MenuServiceTest.java

```diff
@@ -0,0 +1,232 @@
+    void createNoMenuProductsThrowsIllegalArgumentException() {
+        // Arrange
+        Menu request = createMenu("반반치킨", BigDecimal.valueOf(16000), UUID.randomUUID(), true, Collections.emptyList());
+
+        when(menuGroupRepository.findById(any())).thenReturn(Optional.of(new MenuGroup()));
+
+        // Act & Assert
+        assertThatThrownBy(() -> menuService.create(request))
+                .isInstanceOf(IllegalArgumentException.class);
+        verify(menuRepository, never()).save(any(Menu.class));
+    }
+
+    @Test
+    @DisplayName("메뉴 등록 - 상품과 메뉴 상품의 개수가 일치하지 않을 때 IllegalArgumentException 발생")
+    void createMismatchedProductCountsThrowsIllegalArgumentException() {
+        // Arrange
+        MenuGroup menuGroup = createMenuGroup("두마리 메뉴");
+        MenuProduct menuProduct = createMenuProduct(UUID.randomUUID(), 2);  // 두 개의 메뉴 상품
+        Menu request = createMenu("상품과 메뉴 상품의 개수가 일치하지 않는 메뉴", BigDecimal.valueOf(30000), UUID.randomUUID(), true, List.of(menuProduct));
+
+        when(menuGroupRepository.findById(any())).thenReturn(Optional.of(menuGroup));
+        when(productRepository.findAllByIdIn(any())).thenReturn(new ArrayList<>());  // 상품 리스트는 비어있음
+
+        // Act & Assert
+        assertThatThrownBy(() -> menuService.create(request))
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    @DisplayName("메뉴 등록 - 메뉴 가격이 포함된 상품 가격 총합보다 높을 때 IllegalArgumentException 발생")
+    void createMenuPriceGreaterThanProductPriceThrowsIllegalArgumentException() {
+        // Arrange
+        MenuProduct menuProduct = createMenuProduct(UUID.randomUUID(), 1);
+        Menu request = createMenu("통구이", BigDecimal.valueOf(32000), UUID.randomUUID(), true, List.of(menuProduct));
+
+        Product product = createProduct("통구이", BigDecimal.valueOf(16000));
+        when(productRepository.findById(any())).thenReturn(Optional.of(product));
+
+        MenuGroup menuGroup = createMenuGroup("한마리 메뉴");
+        when(menuGroupRepository.findById(any())).thenReturn(Optional.of(menuGroup));
+
+        product = createProduct("통구이", BigDecimal.valueOf(16000));
+        when(productRepository.findAllByIdIn(any())).thenReturn(List.of(product));
+
+        // Act & Assert
+        assertThatThrownBy(() -> menuService.create(request))
+                .isInstanceOf(IllegalArgumentException.class);
+        verify(menuRepository, never()).save(any(Menu.class));
+    }
+
+    @Test
+    @Disabled // 테스트실패 케이스로 버그수정 필요
+    @DisplayName("메뉴 가격 수정 - 메뉴의 상품 가격 총합보다 낮은 가격으로 수정할 때 수정된 메뉴 반환 - 테스트실패")
+    void changePricePriceLessThanProductTotalChangedMenu() {
+        // Arrange
+        Product product1 = createProduct("간장치킨", BigDecimal.valueOf(16000));
+        MenuProduct menuProduct1 = createMenuProduct(product1.getId(), 1, product1);
+        Product product2 = createProduct("순살치킨", BigDecimal.valueOf(16000));
+        MenuProduct menuProduct2 = createMenuProduct(product2.getId(), 1, product2);
```

Comment:
```suggestion
        Product 간장치킨 = createProduct("간장치킨", BigDecimal.valueOf(16000));
        MenuProduct 간장치킨_메뉴상품 = createMenuProduct(product1.getId(), 1, product1);
        Product 순살치킨 = createProduct("순살치킨", BigDecimal.valueOf(16000));
        MenuProduct 순살치킨_메뉴상품 = createMenuProduct(product2.getId(), 1, product2);
```
테스트 코드상 변수명에 넘버링을 하기 보단 직관적으로 정의하는것도 좋은 방법입니다 😄

Reply:
코드에서 한글 변수명을 지정하는게 조금 어색하긴 한데  테스트코드의 직관성 및 가독성이 훨씬 좋아 지는 것 같습니다.
도입해 보겠습니다.